Description: Make sure we use FORTIFY_SOURCE=3 as defined in the dpkg script.
 This fixes FTBFS
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2023-12-27

Index: onetbb/cmake/compilers/Clang.cmake
===================================================================
--- onetbb.orig/cmake/compilers/Clang.cmake
+++ onetbb/cmake/compilers/Clang.cmake
@@ -68,7 +68,7 @@ endif()
 
 set(TBB_COMMON_LINK_LIBS ${CMAKE_DL_LIBS})
 
-set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} $<$<NOT:$<CONFIG:Debug>>:-D_FORTIFY_SOURCE=2>)
+set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} $<$<NOT:$<CONFIG:Debug>>:-D_FORTIFY_SOURCE=3>)
 
 if (MINGW)
     list(APPEND TBB_COMMON_COMPILE_FLAGS -U__STRICT_ANSI__)
Index: onetbb/cmake/compilers/GNU.cmake
===================================================================
--- onetbb.orig/cmake/compilers/GNU.cmake
+++ onetbb/cmake/compilers/GNU.cmake
@@ -75,7 +75,7 @@ set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMO
 if (NOT APPLE AND NOT MINGW)
     set(TBB_LIB_LINK_FLAGS ${TBB_LIB_LINK_FLAGS} -Wl,-z,relro,-z,now,-z,noexecstack)
 endif()
-set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} $<$<NOT:$<CONFIG:Debug>>:-D_FORTIFY_SOURCE=2> )
+set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} $<$<NOT:$<CONFIG:Debug>>:-D_FORTIFY_SOURCE=3> )
 
 
 # TBB malloc settings
Index: onetbb/cmake/compilers/Intel.cmake
===================================================================
--- onetbb.orig/cmake/compilers/Intel.cmake
+++ onetbb/cmake/compilers/Intel.cmake
@@ -21,7 +21,7 @@ if (MSVC)
 elseif (APPLE)
     include(${CMAKE_CURRENT_LIST_DIR}/AppleClang.cmake)
     set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -fstack-protector -Wformat -Wformat-security
-                                 $<$<NOT:$<CONFIG:Debug>>:-fno-omit-frame-pointer -qno-opt-report-embed -D_FORTIFY_SOURCE=2>)
+                                 $<$<NOT:$<CONFIG:Debug>>:-fno-omit-frame-pointer -qno-opt-report-embed -D_FORTIFY_SOURCE=3>)
     set(TBB_OPENMP_FLAG -qopenmp)
     set(TBB_IPO_COMPILE_FLAGS $<$<NOT:$<CONFIG:Debug>>:-ipo>)
 else()
